File: product_management/product_management/src/main/java/com/example/product_management/Service/ProductService.java

package com.example.product_management.Service;

import com.example.product_management.Model.Product;

import java.util.List;

public interface ProductService {

    public Product saveProduct(Product product);
    public List<Product> getAllProduct();
    public Product getProductById(Integer id);
    public String deleteProduct(Integer id);
    public Product updateProduct(Product product,Integer id);
}

File: product_management_ui/src/service/product.service.js
import axios from 'axios';

const APIURL="http://localhost:8081";

class ProductService{
    saveProduct(product){
        return axios.post(APIURL+"/saveProduct",product);
    }

    getAllProduct(){
        return axios.get(APIURL+"/");
    }

    getProductById(id){
        return axios.get(APIURL+"/"+id);
    }

    deleteProduct(id){
        return axios.delete(APIURL+"/deleteProduct/"+id);
    }

    updateProduct(product){
        return axios.put(APIURL+"/updateProduct/"+product.id,product);
    }

}

export default new ProductService;